packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -45,25 +46,53 @@ def draw(self):
             self.figure.dpi = orig_dpi
             self._idle_scheduled = False
 
+    def print_png(self, filename_or_obj, *args,
+                  metadata=None, pil_kwargs=None,
+                  **kwargs):
+
+        from matplotlib import _png
+
+        if metadata is None:
+            metadata = {}
+        metadata = {
+            "Software":
+                f"matplotlib version{__version__}, http://matplotlib.org/",
+            **metadata,
+        }
+
+        canvas = self.get_element('canvas')
+        ctx = canvas.getContext("2d")
+        canvas_data = ctx.getImageData(0, 0, int(ctx.width),
+                                       int(ctx.height)).data
+        data = np.reshape(np.uint8(np.asarray(canvas_data)),
```

Comment:
I think you can do `np.asarray(canvas_data, dtype=np.uint8)` to avoid a potential extra conversion.

Reply:
Sure, That makes sense :)